lightningd/channel_control.c

```diff
@@ -116,6 +116,38 @@ static void peer_got_funding_locked(struct channel *channel, const u8 *msg)
 		lockin_complete(channel);
 }
 
+static void peer_got_announcement(struct channel *channel, const u8 *msg)
+{
+	secp256k1_ecdsa_signature *remote_ann_node_sig;
+	secp256k1_ecdsa_signature *remote_ann_bitcoin_sig;
+
+	remote_ann_node_sig = tal(tmpctx, secp256k1_ecdsa_signature);
+	remote_ann_bitcoin_sig = tal(tmpctx, secp256k1_ecdsa_signature);
```

Comment:
You can allocate them directly off of `channel` if you `tal_free` them on error as well (which should be rare anyway).

Reply:
Yes, now I don't allocate them directly.